src/selectors/identity.js

```diff
@@ -0,0 +1,74 @@
+import { Map as IMap, List as IList } from "immutable";
+import { createSelector } from "reselect";
+
+export const selectIdentities = (state) => state.identity.getIn(["personalInformation", "identities", "byId"], IMap());
+
+export const selectIdentityObj = (state) => state.identity.getIn(["personalInformation", "identities"], IMap())
+
+export const selectActiveIdentity = (state) =>
+  state.identity.getIn(["personalInformation", "activeIdentity"], IMap());
```

Comment:
So this logic should use createSelector to 
const selectActiveCategory = createSelector(
  (state) => state.personalInformation.activeIdentity,
  (state) => state.personalInformation.identities 
  (activeIdentity, identities) => identities.byId[activeIdentity])
);
Adopt just for immutable syntax. The intention is obvious. The same for other stuff. You store references to the entities tables in your store and then cache/memoize them in your selectors.